src/lib/registration/sw_mgmt.rb

```diff
@@ -120,25 +120,29 @@ def self.check_repositories
       true
     end
 
-    # Prepare a pkg-binding product hash using the first base product available
-    # as a template and with the given self_update_id as the product name.
+    # Prepare a pkg-binding product info hash
     #
-    # With a media containing multiple products it is expected that all the
-    # products use the same version and arch.
+    # If there is available products in the media, the first of them will be used as a template,
+    # keeping the self_update_id as the product name, since it is expected that all of them use the
+    # same version and arch.
     #
     # @param self_update_id [String] product name to be used for get the installer updates
-    # @return [Hash,nil] with pkg-binding format; return nil if the
-    # given self_update_id is empty or there is no base product available
-    def self.installer_update_base_product(self_update_id)
-      return if self_update_id.empty?
+    # @param self_update_version [String] product version to be used for get the installer updates
+    #
+    # @return [Hash,nil] with pkg-binding format; nil if the given self_update_id is empty or there
+    #                    is no base product available
+    def self.installer_update_base_product(self_update_id, self_update_version)
       base_product = Y2Packager::Product.available_base_products.first
-      return unless base_product
 
-      # filter out not needed data
+      name = self_update_id.to_s
+      version = base_product ? version_without_release(base_product) : self_update_version.to_s
+
+      return if [name, version].any?(&:empty?)
```

Comment:
This looks too complicated to me, IMHO simple

```suggestion
      return if name.empty? || version.empty?
```

looks much more readable to me. Sure, for larger lists it would make sense, but for just two values I feel the meaning is more obvious.